src/libraries/Microsoft.Extensions.Primitives/src/StringSegment.cs

```diff
@@ -102,7 +102,7 @@ public StringSegment(string buffer, int offset, int length)
                     ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.index);
                 }
 
-                return Buffer[Offset + index];
+                return Buffer![Offset + index];
```

Comment:
I'd prefer
```suggestion
                Debug.Assert(Buffer is not null);
                return Buffer[Offset + index];
```
because if in the ctor a bug get introduced that doesn't set the correct Length when buffer = null, then it will blow up here.